client/task-list/list.js

```diff
@@ -52,20 +52,26 @@ export class TaskList extends Component {
 	possiblyCompleteTaskList() {
 		const { isComplete, name = 'task_list', updateOptions } = this.props;
 		const taskListVariableName = `woocommerce_${ name }_complete`;
-		const taskListToComplete = isComplete
-			? { [ taskListVariableName ]: 'no' }
-			: { [ taskListVariableName ]: 'yes' };
-		if ( name === 'task_list' ) {
-			taskListToComplete.woocommerce_default_homepage_layout =
-				'two_columns';
-		}
+		const remainingIncompleteTasks = !! this.getIncompleteTasks().length;
 
 		if (
-			( ! this.getIncompleteTasks().length && ! isComplete ) ||
-			( this.getIncompleteTasks().length && isComplete )
+			( remainingIncompleteTasks && isComplete ) ||
+			( ! remainingIncompleteTasks && ! isComplete )
 		) {
+			const layoutUpdate =
+				name === 'task_list'
+					? { woocommerce_default_homepage_layout: 'two_columns' }
+					: {};
+
+			const taskListHiddenUpdate =
+				name === 'task_list' && isComplete
+					? { woocommerce_task_list_hidden: 'yes' }
+					: {};
+
 			updateOptions( {
-				...taskListToComplete,
+				[ taskListVariableName ]: isComplete ? 'yes' : 'no',
```

Comment:
Same here, it should be 
```
[ taskListVariableName ]: ! isComplete ? 'yes' : 'no',
```

Reply:
Yeah when I wrote this code, I didn't understand that the original code was toggling the value. I missed that subtlety which also makes me think I should unit test this here.

 It sort of seems odd to do a toggle here in my opinion. What we care most about is whether the task list was just completed, not the original value. The only thing we care about the original value for is to determine if we should re-update the value if its already been updated... (which is covered in the `if`)
 
 Do you think it'd be ok to change this to reflect whether `remainingIncompleteTasks` was true or false instead?